### PR TITLE
rm seed from correlated GP

### LINKED ISSUE #932: Potential bug: Incorrect substitution of device for random seed in GaussianProcessCorrelated
### Description

All tuning iterations of GaussianProcessCorrelated (RBF and Matern32) fail with a gpytorch error:
`Num input variables: 1
Num cases: 100
Num output variables: 72
INFO    2025-11-17 14:20:27,880 - autoemulate - Comparing ['GaussianProcessCorrelatedMatern32', 'GaussianProcessCorrelatedRBF']
INFO    2025-11-17 14:20:27,881 - autoemulate - Running Model: GaussianProcessCorrelatedMatern32: 1/2 (attempt 1/3)
DEBUG   2025-11-17 14:20:27,882 - autoemulate - Running tuner for model "GaussianProcessCorrelatedMatern32"
DEBUG   2025-11-17 14:20:27,882 - autoemulate - Cross-validation parameters: KFold(n_splits=10, random_state=None, shuffle=False)
DEBUG   2025-11-17 14:20:27,883 - autoemulate - Cross-validation split 0: 72 train samples, 8 validation samples
WARNING 2025-11-17 14:20:27,886 - autoemulate - Failed tuning iteration 0 with model params: {"epochs":100,"lr":0.5,"likelihood_cls":"<class 'gpytorch.likelihoods.multitask_gaussian_likelihood.MultitaskGaussianLikelihood'>","scheduler_cls":"<class 'torch.optim.lr_scheduler.LRScheduler'>","scheduler_params":{"policy":"ReduceLROnPlateau","patience":5,"factor":0.5}}: Cannot cast scalar from dtype('O') to dtype('int64') according to the rule 'safe'`

This appears to be because the `device` is passed as a positional argument in `autoemulate.emulators.gaussian_process.exact.create_gp_subclass().GaussianProcessSubclass()`'s call to the superclass constructor: 
`super().__init__(
                x,
                y,
                standardize_x,
                standardize_y,
                likelihood_cls,
                mean_module_fn,
                covar_module_fn,
                fixed_mean_params,
                fixed_covar_params,
                posterior_predictive,
                epochs,
                lr,
                early_stopping,
                device,
                **scheduler_params,
            )`

but `GaussianProcessCorrelated()`'s constructor expects a random seed before the device:
`    # TODO: refactor the init as similar to exact GP base class
    def __init__(
        self,
        x: TensorLike,
        y: TensorLike,
        standardize_x: bool = False,
        standardize_y: bool = True,
        likelihood_cls: type[MultitaskGaussianLikelihood] = MultitaskGaussianLikelihood,
        mean_module_fn: MeanModuleFn = constant_mean,
        covar_module_fn: CovarModuleFn = rbf_plus_constant,
        fixed_mean_params: bool = False,
        fixed_covar_params: bool = False,
        posterior_predictive: bool = False,
        epochs: int = 50,
        lr: float = 2e-1,
        early_stopping: EarlyStopping | None = None,
        seed: int | None = None,
        device: DeviceLike | None = None,
        scheduler_cls: type[LRScheduler] | None = None,
        scheduler_params: dict | None = None,
    ):`

This causes numpy to throw an error.

### Reproduction steps

```bash
1. Instantiate an AutoEmulate object with either `gaussianprocesscorrelatedrbf` or `gaussianprocesscorrelatedmatern32` models on a CPU-only architecture
2. Observe bug
```

### Version

Tried on the latest release (1.2.0) and cloned from `main`

### Screenshots

```bash

```

### OS

Linux

### PR DESCRIPTION
**Pull request by radka-j**
Closes #932

### CODE REVIEW

**Review by sgreenbury — APPROVED** [review]
LGTM, thanks!
